src/librustc/hir/lowering.rs

```diff
@@ -862,9 +865,14 @@ impl<'a> LoweringContext<'a> {
                 PatKind::Wild => hir::PatKind::Wild,
                 PatKind::Ident(ref binding_mode, pth1, ref sub) => {
                     self.with_parent_def(p.id, |this| {
+                        let name = match this.resolver.get_resolution(p.id).map(|d| d.full_def()) {
+                            // Only pattern bindings are renamed
```

Comment:
I tables of "canonical bindings" were filled during the main name resolution pass and saved for later, then HIR could keep plain non-renamed `Name`s for everything.